### PR TITLE
Minor improvements to schema parser

### PR DESCRIPTION
**Pull request by elle-j**
## What, How & Why?

Some minor refactors and renaming in the schema parser have been made.

### CODE REVIEW

File: packages/realm/src/schema/normalize.ts [hunk 1]
```diff
@@ -72,6 +72,8 @@ const COLLECTION_SHORTHAND_TO_NAME: Readonly<Record<string, string>> = {
   "<>": "set",
 };
 
+const COLLECTION_SUFFIX_LENGTH = 2;
```
Comment (kneth) [inline packages/realm/src/schema/normalize.ts:75]:
```suggestion
const COLLECTION_SUFFIX_LENGTH = "[]".length;
```

**Review by kneth — APPROVED** [review]
Definitely better naming.

File: packages/realm/src/schema/normalize.ts [hunk 2]
```diff
@@ -353,42 +354,52 @@ function optionalIsImplicitlyTrue(type: string, objectType: string | undefined):
...
+  if (hasCollectionSuffix(input)) {
+    shorthands.push(input.substring(input.length - COLLECTION_SUFFIX_LENGTH));
+    input = input.substring(0, input.length - COLLECTION_SUFFIX_LENGTH);
+  }
+  if (input.endsWith("?")) {
+    shorthands.push("?");
+  }
+
+  return shorthands;
+}
```
Comment (kraenhansen) [inline packages/realm/src/schema/normalize.ts:401]: Great candidate for a separate function 👍

**Review by kraenhansen — APPROVED** [review]

File: packages/realm/src/schema/normalize.ts (hunk 2 above)
Comment (elle-j) [reply packages/realm/src/schema/normalize.ts:401]: Indeed, thanks 🙂

**Pull request merged**